weaknesses/MASVS-CODE/MASWE-0076.md

```diff
@@ -1,38 +1,79 @@
+- **Regulatory and Policy Non-Compliance**: Including dependencies with publicly known CVEs may violate regulatory requirements (e.g., GDPR, HIPAA, PCI-DSS) or platform security policies (e.g., Google Play or App Store guidelines). Failure to update or remediate such vulnerabilities can result in app rejection, fines, or mandatory disclosures.
 
 ## Modes of Introduction
 
-Mobile apps rely heavily on dependencies. These dependencies can be "closed-source" through vendor products or "open-source" and maintained by the community.
-
-Dependencies can be implemented manually by adding them into the project and linking them, but they are usually added through dependency managers, which handle the integration into the project's files.
+- **Direct Dependencies**: Vulnerable dependencies can be introduced into the app either manually (by copying and linking source or binary files) or more commonly via package managers and build tools (e.g., Gradle, CocoaPods, Swift Package Manager). This includes both first- and third-party SDKs, and may involve both statically and dynamically linked libraries.
+- **Transitive Dependencies**: Dependencies can be pulled in indirectly through other libraries or SDKs that the app uses. This means that an app may still be affected by a vulnerable library if one of its dependencies includes it, even if the app does not directly include the library.
+- **Dynamically Loaded Dependencies**: Some libraries may be dynamically loaded at runtime, which can make it difficult to track and manage dependencies. This can lead to situations where a vulnerable version of a library is used without the developer's knowledge.
+- **Outdated Platform Security Components**: Mobile apps may depend on platform-provided security components, such as cryptographic libraries or SSL/TLS implementations. If these components are outdated or lack timely updates, they can introduce known vulnerabilities into the application. For instance, on Android, the system's security provider responsible for secure network communications must be explicitly updated by the developer at app startup.
 
 ## Mitigations
 
-- **Keep Dependencies Updated**: Regularly update dependencies to their latest secure versions to ensure that any known vulnerabilities are patched.
-- **Regular Dependency Audits:**: Continuously scan and audit third-party libraries for vulnerabilities using Software Composition Analysis (SCA) tools in the CI/CD pipeline, like @MASTG-TOOL-0131 or @MASTG-TOOL-0132.
-- **Software Bill of Material (SBOM)**: Create a SBOM and manage the dependencies by using tools like @MASTG-TOOL-0134 and @MASTG-TOOL-0132.
-- **Remove Unused Dependencies**: Regularly review and remove any unused or unnecessary libraries to reduce the app's attack surface.
+- **Keep Dependencies Up to Date**: Regularly update all third-party libraries and SDKs to their latest stable versions to ensure known vulnerabilities are patched.
```

Comment:
I don't really agree here. You should pin your version and you don't have to update unless a security vulnerability was identified. So they should monitor the updates of all used dependencies and update in case it is a security update.
```suggestion
- **Monitor Release Notes of All Used Dependencies**: Monitor the release notes of all used dependencies to identify security-critical updates. In case such an update is available, the dependency should be updated to the latest stable version that fixes the vulnerability. Otherwise, dependencies should be pinned to a specific version to prevent supply-chain attacks.
```

Reply:
I see your point, what about...?

```suggestion
- **Update Dependencies Responsibly**: Regularly monitor all used third-party dependencies for security-related updates. Apply updates when they fix known vulnerabilities, and pin versions explicitly to prevent unexpected changes or supply-chain attacks.
```